consul-lambda-registrator/extension/extension_test.go

```diff
@@ -0,0 +1,126 @@
+package extension_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul/tlsutil"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/require"
+
+	ext "github.com/hashicorp/terraform-aws-consul-lambda-registrator/consul-lambda-registrator/extension"
+)
+
+func TestExtension(t *testing.T) {
```

Comment:
I would be very tempted to only test this in an integration test. Nice work putting together a lower level test for this.